packages/deployments/contracts/contracts_forge/core/connext/helpers/RelayerProxyHub.t.sol

```diff
@@ -0,0 +1,483 @@
+    vm.prank(OWNER);
+    vm.expectEmit(true, true, true, true);
+    emit PropagateCooldownChanged(123, 300);
+    proxy.setPropagateCooldown(123);
+    assertEq(proxy.propagateCooldown(), 123);
+  }
+
+  function test_RelayerProxyHub__setHubConnector_onlyOwner_failsIfNotOwner(address sender) public {
+    vm.assume(sender != OWNER);
+    vm.prank(sender);
+    vm.expectRevert(ProposedOwnable__onlyOwner_notOwner.selector);
+    proxy.setHubConnector(address(123), 123);
+  }
+
+  function test_RelayerProxyHub__setHubConnector_onlyOwner_works() public {
+    vm.prank(OWNER);
+    vm.expectEmit(true, true, true, true);
+    emit HubConnectorChanged(address(123), address(0), 123);
+    proxy.setHubConnector(address(123), 123);
+    assertEq(proxy.hubConnectors(123), address(123));
+  }
+
+  function test_RelayerProxyHub__propagateWorkable_isFalseIfRootIsSame() public {
+    vm.mockCall(
+      address(proxy.rootManager()),
+      abi.encodeWithSelector(IRootManager.dequeue.selector),
+      abi.encode(bytes32(uint256(1)), 123)
+    );
+
+    vm.mockCall(
+      address(proxy.rootManager()),
+      abi.encodeWithSelector(IRootManager.lastPropagatedRoot.selector),
+      abi.encode(vm.parseBytes32("0x0000000000000000000000000000000000000000000000000000000000000001"))
+    );
+
+    assertEq(proxy.propagateWorkable(), false);
+  }
+
+  function test_RelayerProxyHub__propagateWorkable_isTrue() public {
+    vm.mockCall(
+      address(proxy.rootManager()),
+      abi.encodeWithSelector(IRootManager.dequeue.selector),
+      abi.encode(vm.parseBytes32("0x0000000000000000000000000000000000000000000000000000000000000001"), 123)
+    );
+
+    vm.mockCall(
+      address(proxy.rootManager()),
+      abi.encodeWithSelector(IRootManager.lastPropagatedRoot.selector),
+      abi.encode(vm.parseBytes32("0x0000000000000000000000000000000000000000000000000000000000000002"))
+    );
+
+    assertEq(proxy.propagateWorkable(), true);
+  }
+
+  function test_RelayerProxyHub__propagate_failsIfNotGelatoRelayer(address sender) public {
+    vm.assume(sender != _gelatoRelayer);
+    vm.prank(sender);
+    vm.expectRevert(abi.encodeWithSelector(RelayerProxy__onlyRelayer_notRelayer.selector, sender));
+    proxy.propagate(_hubConnectors, _messageFees, _encodedData, _relayerFee);
+  }
+
+  function test_RelayerProxyHub__propagate_works() public {
+    vm.mockCall(address(proxy.rootManager()), abi.encodeWithSelector(IRootManager.propagate.selector), abi.encode());
+    vm.prank(_gelatoRelayer);
+    vm.expectEmit(true, true, true, true);
+    emit FundsDeducted(123 + 456, 1 ether);
+    vm.expectEmit(true, true, true, true);
+    emit FundsDeducted(123, 1 ether - 123);
+    proxy.propagate(_hubConnectors, _messageFees, _encodedData, _relayerFee);
+  }
+
+  function test_RelayerProxyHub__propagateKeep3r_failsIfNotPriority(address sender) public {
+    vm.assume(sender != _autonolas);
+    vm.roll(100);
+    vm.prank(sender);
```

Comment:
can you explain the priority mechanism to me? im not following what this test is testing

Reply:
We have a mechanism for prioritizing based on block number. If the keeper is not the prioritized keeper, it can not send while block number < `_autonolasPriority`